R/FilterState-utils.R

```diff
@@ -75,216 +84,280 @@ init_filter_state <- function(x,
     x = x,
     x_reactive = x_reactive,
-    varname = varname,
-    varlabel = varlabel,
     dataname = dataname,
+    varname = varname,
+    choices = choices,
+    selected = selected,
+    keep_na = keep_na,
+    keep_inf = keep_inf,
+    fixed = fixed,
     extract_type = extract_type
   )
+  args <- append(args, list(...))
+
+  do.call(InteractiveFilterState$new, args)
 }
 
 #' @keywords internal
 #' @export
 init_filter_state.logical <- function(x,
                                       x_reactive = reactive(NULL),
+                                      dataname,
                                       varname,
-                                      varlabel = attr(x, "label"),
-                                      dataname = NULL,
-                                      extract_type = character(0)) {
-  if (is.null(varlabel)) varlabel <- character(0)
-  LogicalFilterState$new(
+                                      choices = unique(na.omit(x)),
+                                      selected = NULL,
+                                      keep_na = NULL,
+                                      keep_inf = NULL,
+                                      fixed = FALSE,
+                                      extract_type = character(0),
+                                      ...) {
+  args <- list(
     x = x,
     x_reactive = x_reactive,
-    varname = varname,
-    varlabel = varlabel,
     dataname = dataname,
+    varname = varname,
+    choices = choices,
+    selected = selected,
+    keep_na = keep_na,
+    keep_inf = keep_inf,
+    fixed = fixed,
     extract_type = extract_type
   )
+  args <- append(args, list(...))
+
+  do.call(LogicalFilterState$new, args)
 }
 
 #' @keywords internal
 #' @export
 init_filter_state.numeric <- function(x,
                                       x_reactive = reactive(NULL),
+                                      dataname,
                                       varname,
-                                      varlabel = attr(x, "label"),
-                                      dataname = NULL,
-                                      extract_type = character(0)) {
-  if (is.null(varlabel)) varlabel <- character(0)
+                                      choices = unique(na.omit(x)),
+                                      selected = NULL,
+                                      keep_na = NULL,
+                                      keep_inf = NULL,
+                                      fixed = FALSE,
+                                      extract_type = character(0),
+                                      ...) {
+  args <- list(
+    x = x,
+    x_reactive = x_reactive,
+    dataname = dataname,
+    varname = varname,
+    choices = choices,
+    selected = selected,
+    keep_na = keep_na,
+    keep_inf = keep_inf,
+    fixed = fixed,
+    extract_type = extract_type
+  )
+  args <- append(args, list(...))
+
   if (length(unique(x[!is.na(x)])) < getOption("teal.threshold_slider_vs_checkboxgroup")) {
```

Comment:
Since we drop NAs here anyway, is there a point to having a default `choices = unique(na.omit(x))`?